File: WareHouse/BartBotV2_DefaultOrganization_20231107121605/game.java
'''
This class represents the game itself. It has methods to start the game, choose a character, and perform actions on the character. It also checks if the game should end after each action.
'''
public class Game {
    private Character character;
    public void start() {
        System.out.println("Welcome to the Simpsons Tamagotchi Game!");
        System.out.println("Please choose your character: Bart, Lisa, Maggie, Homer or Marge");
    }
    public void chooseCharacter(String name) {
        this.character = new Character(name);
        System.out.println("You have chosen " + name + " as your pet.");
    }
    public void performAction(String action) {
        if (action.equals("feed")) {
            this.character.feed();
        } else if (action.equals("play")) {
            this.character.play();
        } else if (action.equals("clean")) {
            this.character.clean();
        }
        if (this.character.isCritical()) {
            System.out.println("Game Over. Your character's condition has reached a critical level.");
            System.exit(0);
        }
    }
}

File: WareHouse/BartBotV2_DefaultOrganization_20231107123608/game.py
"""
This file contains the Game class which controls the game flow.
"""
from __future__ import annotations

from gui import GUI
from simpson_pet import SimpsonPet


class Game:
    def __init__(self):
        self.gui = GUI()
        self.pet = None

    def start(self):
        self.gui.draw_intro()
        pet_name = self.gui.get_user_input(
            "Choose your pet: Bart, Lisa, Maggie, Homer or Marge",
        )
        self.pet = SimpsonPet(pet_name)
        self.main_loop()

    def main_loop(self):
        while self.pet.is_alive():
            self.gui.update_status(self.pet)
            action = self.gui.get_user_input(
                "What do you want to do? (Feed / Play / Exit)",
            )
            while action.lower() not in ["feed", "play", "exit"]:
                print("Invalid action. Please try again.")
                action = self.gui.get_user_input(
                    "What do you want to do? (Feed / Play / Exit)",
                )
            if action.lower() == "feed":
                self.pet.feed()
            elif action.lower() == "play":
                self.pet.play()
            elif action.lower() == "exit":
                break
            self.pet.time_passes()
        self.gui.draw_outro()
